packages/core/src/utils/cache.ts

```
@@ -48,42 +48,67 @@ export class Cache<K, V> {
       return cachedValue as ReturnType<T>;
     }
-    const result = fn(...args);
-    this.set(key, result, ttl);
-    return result;
+    try {
+      const result = await fn(...args);
+      this.set(key, result, ttl);
+      return result;
+    } catch (error) {
+      // Don't cache errors
+      throw error;
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)